### PR TITLE
User note

### PR DESCRIPTION
**Pull request by jnelsonjava**
- Added field to Accounts table
- Updated Account, AccountSummary, and StudyParticipant models
- Updated related controllers/services

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccount.java [hunk 1]
```diff
@@ -125,6 +126,7 @@ public HibernateAccount(DateTime createdOn, String appId, String orgId, String f
         this.id = id;
         this.status = status;
         this.synapseUserId = synapseUserId;
+        this.note = note;
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccount.java]: We can remove note from the AccountRef constructor.

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java [hunk 2]
```diff
@@ -254,6 +254,7 @@ AccountSummary unmarshallAccountSummary(HibernateAccount acct) {
         builder.withSynapseUserId(acct.getSynapseUserId());
         builder.withAttributes(acct.getAttributes());
         builder.withOrgMembership(acct.getOrgMembership());
+        builder.withNote(acct.getNote());
```
Comment (DwayneJengSage) [inline src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java:257]: Let's also add this to HibernateAccountDaoTest.unmarshallAccountSummarySuccess(), to make sure we are exercising this piece of code.

File: src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java [hunk 3]
```diff
@@ -457,6 +458,7 @@ public IdentifierHolder createParticipant(App app, StudyParticipant participant,
         account.setPhoneVerified(FALSE);
         account.setHealthCode(generateGUID());
         account.setStatus(UNVERIFIED);
+        account.setNote(participant.getNote());
```
Comment (jnelsonjava) [inline src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java]: Remove this, it's handled in the update account method.

File: src/test/java/org/sagebionetworks/bridge/hibernate/HibernateAccountTest.java [hunk 4]
```diff
@@ -134,6 +135,7 @@ public void canSerialize() throws Exception {
         assertEquals(node.get("version").intValue(), 1);
         assertEquals(toSet(node, "dataGroups"), ImmutableSet.of("group1", "group2"));
         assertEquals(toSet(node, "languages"), ImmutableSet.of("en", "fr"));
+        assertEquals(node.get("note").textValue(), "note");
```
Comment (jnelsonjava) [inline src/test/java/org/sagebionetworks/bridge/hibernate/HibernateAccountTest.java]: Change "note" text to be different from field name.

**Review by DwayneJengSage — CHANGES_REQUESTED** [review]

**Comment by alxdarksage** [comment]
Still need to add the test suggested by Dwayne.

**Review by alxdarksage — CHANGES_REQUESTED** [review]
Still need to add the test suggested by Dwayne.

File: src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java (hunk 2 above)
Comment (jnelsonjava) [reply src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java:257]: The HibernateAccountDaoTest.unmarshallAccountSummarySuccess() test should be updated now to check that the note field is coming through in the Account Summary.

**Review by alxdarksage — APPROVED** [review]

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**